src/SignalR/clients/ts/signalr/src/Errors.ts

```diff
@@ -158,6 +158,32 @@ export class FailedToStartTransportError extends Error {
     }
 }
 
+/** Error thrown when the negotiation with the server failed to complete. */
+/** @private */
+export class FailedToNegotiateWithServerError extends Error {
```

Comment:
cc/ @BrennanConroy @halter73 just adding this additional ErrorType, similar to what we did before for the Transport errors. Could I get a signoff for this part if it's okay with you?